compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingIntoPackUnPack.cpp

```diff
@@ -246,62 +336,34 @@ static FailureOr<Operation *> lowerOpWithEncoding(
 
   FailureOr<MaterializeEncodingInfo> materializeEncodingInfo =
       materializeEncodingFn(getOriginalTypeWithEncoding(
-          matmulOp.getResultTypes()[0].cast<RankedTensorType>()));
+          linalgOp->getResultTypes()[0].cast<RankedTensorType>()));
+
   Operation *result;
   if (failed(materializeEncodingInfo)) {
-    result = dropEncodingAndCloneOp(rewriter, matmulOp, convertedInputOperands,
-                                    convertedOutputOperands);
+    result = dropEncodingAndCloneOp(rewriter, linalgOp,
+                                    ValueRange{operands[0], operands[1]},
+                                    ValueRange{operands[2]});
```

Comment:
[optional] I think it's safer if we can rewrite it to operands.take_front/take_back/drop_front. E.g.,

```suggestion
                                    operands.take_front(inputs.size()),
                                    operands.drop_front(inputs.size()));
```